src/DeviceInterface.cpp

```diff
@@ -146,35 +198,20 @@ int halide_weak_device_free(void *user_context, struct buffer_t *buf) {
 EXPORT_SYM(halide_weak_device_free)
 
 const struct halide_device_interface_t *halide_cuda_device_interface() {
-    Target target(get_host_target());
-    target.set_feature(Target::CUDA);
-    struct halide_device_interface_t *(*fn)();
-    if (lookup_runtime_routine("halide_cuda_device_interface", target, fn)) {
-        return (*fn)();
-    }
-    return nullptr;
+    Target t = get_jit_target_from_environment().with_feature(Target::CUDA);
```

Comment:
Should this be get_host_target, like it was before? It's a bit sketchy for this API to depend on an environment variable (which get_jit_target_from_environment does). Also, this comment would be obsoleted if we changed get_device_interface_for_device_api above.